### PR TITLE
feat: create a footer component

### PR DESCRIPTION
**Pull request by macszl**
Related work item: GH-6
Closes: #6

### CODE REVIEW

**Comment by macszl** [comment]
This is currently a draft, unfinished version of the footer component. Looking for a better way to handle multiple text styles in one line.

File: components/Footer.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        bgcolor='#1976D2'
+        spacing={0}
+        padding={0}
+        height={158}
+      >
+        <Grid
+          style={{ textAlign: 'center' }}
+          color='#ffffff'
+        >
+          <div style={{ alignItems: 'center', display: 'flex', gap: 4 }}>
```
Comment (kaczor6418) [inline components/Footer.tsx:21]: Styles should be placed in .scss file not inline in JavaScript Code

File: components/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            <Typography sx={{ fontWeight: '500' }}>Copyright © 2022 </Typography>
```
Comment (kaczor6418) [inline components/Footer.tsx:22]: for **fontWeight**, please use function and appropriate value from default theme that we are using:
[default theme](https://mui.com/material-ui/customization/default-theme/)

Example of how to use default theme
```tsx
<Button sx={{ marginBottom: theme => theme.spacing(3)}} />
```

File: components/Footer.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import '@fontsource/roboto';
```
Comment (kaczor6418) [inline components/Footer.tsx:1]: why you import this here ?

File: components/Footer.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+export function Footer() {
+  return (
+    <footer>
+      <Grid
+        container
+        direction='column'
+        alignItems='center'
+        justifyContent='center'
+        bgcolor='#1976D2'
```
Comment (kaczor6418) [inline components/Footer.tsx:12]: use color provided by default theme

File: src/templates/Footer/Footer.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+export function Footer() {
+  const defaultTheme = createTheme();
+  return (
+    <footer>
+      <Grid
+        container
+        direction='column'
+        alignItems='center'
+        justifyContent='center'
+        bgcolor='#1976D2'
```
Comment (kaczor6418) [inline src/templates/Footer/Footer.tsx:15]: use color provided from default theme

**Review by kaczor6418 — CHANGES_REQUESTED** [review]

File: src/templates/Footer/Footer.tsx (hunk 5 above)
Comment (macszl) [reply src/templates/Footer/Footer.tsx]: Should be fixed in 71d21b2

File: components/Footer.tsx (hunk 4 above)
Comment (macszl) [reply components/Footer.tsx]: Should be fixed in 71d21b2

File: components/Footer.tsx (hunk 1 above)
Comment (macszl) [reply components/Footer.tsx]: Design does not use scss files anymore

File: src/templates/Footer/Footer.tsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        spacing={0}
+        padding={0}
+        height={158}
+      >
+        <Grid
+          style={{ textAlign: 'center' }}
+          color={defaultTheme.palette.common.white}
+        >
+          <div className='spacer'>
```
[the remark this answers is not in the document]
Comment (macszl) [reply src/templates/Footer/Footer.tsx]: Footer does not use css classes anymore.

File: src/templates/Footer/Footer.scss [hunk 7]
```diff
@@ -0,0 +1,6 @@
+.spacer
+{
+    align-items: center;
+    display: flex;
+    gap: 4px;
```
[the remark this answers is not in the document]
Comment (macszl) [reply src/templates/Footer/Footer.scss]: Footer does not use css classes anymore, since 71d21b2

File: components/Footer.tsx (hunk 3 above)
Comment (macszl) [reply components/Footer.tsx]: Typography docs say that import '@fontsource/roboto' is necessary to use the Roboto font.
https://mui.com/material-ui/customization/typography/

To avoid import duplication, I've moved the import statement to main in 037002d.

File: src/templates/Footer/Footer.tsx [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+const defaultTheme = createTheme({
+  breakpoints: {
+    values: {
+      base: 0,
+      desktop: 1024,
+      mobile: 360,
+      tablet: 768,
+    },
+  },
+});
```
Comment (kaczor6418) [inline src/templates/Footer/Footer.tsx:27]: This should be in a separated file + you should pass our customTheme into root theme provider. As this breakpoint will be used across the whole application:
https://mui.com/material-ui/customization/theming/#custom-variables

File: tsconfig.json [hunk 9]
```diff
@@ -30,5 +30,5 @@
     "forceConsistentCasingInFileNames": true,
     "jsx": "react-jsx"
   },
-  "include": ["./src"]
+  "include": ["src/**/*"]
```
Comment (kaczor6418) [inline tsconfig.json:33]: Why did you change this?

File: src/App.tsx [hunk 10]
```diff
@@ -1,3 +0,0 @@
-export function App() {
-  return <div className='App'></div>;
-}
```
Comment (kaczor6418) [inline src/App.tsx:3]: `App.tsx` shouldn't be removed. We should keep this because this is our root component.

We should use `react-router` to define routing path and display appropriate pages for given path:

https://v5.reactrouter.com/web/example/route-config

File: src/main.tsx [hunk 11]
```diff
@@ -1,10 +1,11 @@
...
+import './main.scss';
+import '@fontsource/roboto';
 import * as React from 'react';
 import * as ReactDOM from 'react-dom/client';
-import { App } from './App';
+import { Footer } from './templates/Footer/Footer';
 
 ReactDOM.createRoot(document.getElementById('root') as HTMLElement).render(
   <React.StrictMode>
-    <App />
```
Comment (kaczor6418) [inline src/main.tsx:8]: we should keep `App.tsx`

This is also a place where you should use  `<ThemeProvider>` to inject our custom theme

File: src/App.tsx (hunk 10 above)
Comment (kaczor6418) [reply src/App.tsx:3]: See how to use react-router with mui:

https://stackoverflow.com/a/63221012/9095988

**Review by kaczor6418 — CHANGES_REQUESTED** [review]
Almost there

File: src/main.tsx (hunk 11 above)
Comment (macszl) [reply src/main.tsx:8]: Should be fixed in 2b03974

File: tsconfig.json (hunk 9 above)
Comment (macszl) [reply tsconfig.json]: Changed in 2b03974

File: src/templates/Footer/Footer.tsx (hunk 8 above)
Comment (macszl) [reply src/templates/Footer/Footer.tsx]: Changed in 2b03974

File: src/components/IconsGroup/IconsGroup.tsx [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+        <ImportContacts> </ImportContacts>
+      </Link>
+
+      <Link
+        target='_blank'
+        href='https://github.com/dorm-social-room-manager'
+        color='inherit'
+      >
+        <GitHub></GitHub>
+      </Link>
```
Comment (kaczor6418) [inline src/components/IconsGroup/IconsGroup.tsx:29]: You should get icon-list with URLs as parameter and then use this to render icons. With this approach, where you are hard-coding links with icons, we won't be able to use in other components if we will need different icons

**Review by kaczor6418 — CHANGES_REQUESTED** [review]

File: src/templates/Footer/Footer.tsx [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+        <div>
+          <Typography
+            sx={{
+              display: 'inline',
+              fontWeight: (theme) => {
+                return theme.typography.fontWeightMedium;
+              },
+            }}
+          >
+            Copyright © 2022
```
Comment (kaczor6418) [inline src/templates/Footer/Footer.tsx:31]: We should use i18n for translations

File: src/templates/Footer/Footer.tsx [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+              },
+              textDecoration: 'underline',
+            }}
+          >
+            <Link
+              href='https://github.com/dorm-social-room-manager/dsrm-frontend/blob/master/LICENSE.MD'
+              target='_blank'
+              color='inherit'
+            >
+              Policy terms
```
[the remark this answers is not in the document]
Comment (macszl) [reply src/templates/Footer/Footer.tsx]: Added in 92fa2b8

File: src/templates/Footer/Footer.tsx (hunk 13 above)
Comment (macszl) [reply src/templates/Footer/Footer.tsx]: Added in 92fa2b8

File: src/i18n.ts [hunk 15]
```diff
@@ -0,0 +1,29 @@
...
+  en,
+  pl,
+};
+
+export const availableLanguages = Object.keys(resources);
+
+void i18n
+  .use(Backend)
+  .use(LanguageDetector)
+  .use(initReactI18next)
```
Comment (kaczor6418) [inline src/i18n.ts:18]: I think we do not this right now

File: src/i18n.ts [hunk 16]
```diff
@@ -0,0 +1,29 @@
...
+  .init({
+    debug: true,
```
Comment (kaczor6418) [inline src/i18n.ts:20]: I think we do not need this `debug: true`

File: src/i18n.ts [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+import i18n from 'i18next';
+import { initReactI18next } from 'react-i18next';
+import LanguageDetector from 'i18next-browser-languagedetector';
+
+const resources = {
+  en,
+  pl,
+};
+
+export const availableLanguages = Object.keys(resources);
```
Comment (kaczor6418) [inline src/i18n.ts:13]: It looks like it is unused

File: src/locales/Footer/en.json [hunk 18]
```diff
@@ -0,0 +1,7 @@
+{
+    "common":{
+        "copyright":"Copyright © 2022",
+        "authors": "DSRM Team.",
+        "policy": "Policy terms"
+    }
+}
```
Comment (kaczor6418) [inline src/locales/Footer/en.json:7]: do not wrap keys in `common` just directly define keys just to limit nesting

File: src/locales/Footer/pl.json [hunk 19]
```diff
@@ -0,0 +1,7 @@
+{
+    "common":{
+        "copyright":"Copyright © 2022",
+        "authors": "Zespół DSRM.",
+        "policy": "Policy terms"
+    }
+}
```
Comment (kaczor6418) [inline src/locales/Footer/pl.json:7]: do not wrap keys in common just directly define keys just to limit nesting

File: src/react-i18next.d.ts [hunk 20]
```diff
@@ -0,0 +1,9 @@
+import 'react-i18next';
+import en from './locales/Footer/en.json';
+
+declare module 'react-i18next' {
+  interface CustomTypeOptions {
+    defaultNS: 'common';
+    resources: typeof en;
+  }
+}
```
Comment (kaczor6418) [inline src/react-i18next.d.ts:9]: let's skip TS configuration for now, I will create separate ticket for this

>Get rid of this file as it doesn't change anything for now

File: tsconfig.json [hunk 21]
```diff
@@ -28,7 +28,9 @@
     "emitDecoratorMetadata": true,
     "skipLibCheck": true,
     "forceConsistentCasingInFileNames": true,
-    "jsx": "react-jsx"
+    "jsx": "react-jsx",
+    "resolveJsonModule": true,
+    "esModuleInterop": true,
```
Comment (kaczor6418) [inline tsconfig.json:33]: `esModuleInterop` is not needed in my opinion

File: src/templates/Footer/Footer.tsx [hunk 22]
```diff
@@ -0,0 +1,69 @@
+import '../../i18n';
```
Comment (kaczor6418) [inline src/templates/Footer/Footer.tsx:1]: this should not be imported here. Please import i18n in `main.tsx` as i18n should start initialization in root of our application

File: src/locales/Footer/en.json (hunk 18 above)
Comment (kaczor6418) [reply src/locales/Footer/en.json:7]: let's put this into `locales/en/footer.json`

File: src/locales/Footer/pl.json (hunk 19 above)
Comment (kaczor6418) [reply src/locales/Footer/pl.json:7]: let's put this into `locales/pl/footer.json`

File: package.json [hunk 23]
```diff
@@ -19,9 +19,13 @@
   "dependencies": {
     "@emotion/react": "^11.10.5",
     "@emotion/styled": "^11.10.5",
+    "@fontsource/roboto": "^4.5.8",
+    "@mui/icons-material": "^5.10.16",
     "@mui/material": "^5.10.15",
     "formik": "^2.2.9",
     "i18next": "^22.0.6",
+    "i18next-browser-languagedetector": "^7.0.1",
+    "i18next-http-backend": "^2.1.0",
```
Comment (kaczor6418) [inline package.json:28]: we do not need this

**Review by kaczor6418 — CHANGES_REQUESTED** [review]

File: package.json (hunk 23 above)
Comment (macszl) [reply package.json]: Changed in 16f4545

File: src/i18n.ts (hunk 15 above)
Comment (macszl) [reply src/i18n.ts]: Changed in 16f4545

File: src/i18n.ts (hunk 16 above)
Comment (macszl) [reply src/i18n.ts]: Changed in 16f4545

File: src/components/IconsGroup/IconsGroup.tsx (hunk 12 above)
Comment (macszl) [reply src/components/IconsGroup/IconsGroup.tsx]: Resolved in c2c7072

**Review by kaczor6418 — APPROVED** [review]
LGTM 😎
